girder/events.py

```diff
@@ -317,8 +316,12 @@ def trigger(eventName, info=None, pre=None, async=False, daemon=False):
 
 _deprecated = {}
 _mapping = {}
+daemon = ForegroundEventsDaemon()
+
 
-if config.getConfig()['server'].get('disable_event_daemon', False):
-    daemon = ForegroundEventsDaemon()
-else:
-    daemon = AsyncEventsThread()
+def setupDaemon():
+    global daemon
+    if config.getConfig()['server'].get('disable_event_daemon', False):
+        daemon = ForegroundEventsDaemon()
```

Comment:
Since the synchronous event daemon is defined on import, should this just be a noop? Otherwise at run-time the import-time daemon is clobbered?

Reply:
The idea here is for this method to be idempotent since it's doing stuff at server startup and affects stuff in module scope. We could do some type checking, but these are light objects and they'll be garbage collected, so I wasn't worried about the clobbering.

Reply:
My concern about the clobbering is that any downstream binding to events at import time (rather than in a `load` function for instance) would have their bindings removed, right? (I don't think we should necessarily accomodate that, just checking).

Reply:
No, that would be fine, the event bindings live in module scope under the `_mapping` global. The event daemons are stateless in that regard.